src/Chapter2.hs

```diff
@@ -842,7 +867,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate x l
+    | x < 0 = []
+    | otherwise = (drop x l) ++ (take x l)
```

Comment:
Very interesting solution 🎨 👏🏼 

Although it won't cover cases when the rotation argument is bigger than the list's length. 

You can use `take` with the `cycle` function as an alternative solution 🙂